### PR TITLE
Allow using lactose-free milk in recipes

### PR DESCRIPTION
**Pull request by AtomicFox556**
#### Summary
Content "Allow using lactose-free milk in recipes"

#### Purpose of change
Allow preparing food with plant-based lactose-free milk, in case actual milk is unavailable.

#### Describe the solution
Add soy and almond milk as possible milk substitutes for recipes in which it'd make sense. Also allow using condensed milk and powdered milk in some recipes where it'd make sense (mostly food that is sweet).

#### Describe alternatives you've considered
Making a lactose-free variant of every edible item which can end up having milk in it, but this results in a huge amount of new items added; probably better to represent whether an item contains lactose or not with a flag, just like it's done for food items made from human flesh.

#### Testing
Launched the game with those changes, made sure that no errors happen and that new milk options show up in recipes.

#### Additional context
None.

### CODE REVIEW

**Comment by Zireael07** [comment]
Which items are those? Plant milk doesn't have lactose, so it's unsuitable for e.g. yoghurt IIRC

**Comment by AtomicFox556** [comment]
> Which items are those? Plant milk doesn't have lactose, so it's unsuitable for e.g. yoghurt IIRC

Baking recipes and drinks, mostly; nothing that make uses of fermentation like cheese or yoghurt as you mentioned.

**Comment by AtomicFox556** [comment]
> > Which items are those? Plant milk doesn't have lactose, so it's unsuitable for e.g. yoghurt IIRC
> 
> Baking recipes and drinks, mostly; nothing that makes use of fermentation like cheese or yoghurt as you mentioned.

**Comment by AtomicFox556** [comment]
No idea why the item info tests are failing; everything looks fine to me.
image

**Comment by AtomicFox556** [comment]
Figured it out, I think.

File: data/json/recipes/food/frozen.json [hunk 1]
```diff
@@ -170,7 +176,7 @@
     "qualities": [ { "id": "COOK", "level": 3 } ],
     "tools": [ [ [ "rock_quern", -1 ], [ "clay_quern", -1 ], [ "mortar_pestle", -1 ], [ "food_processor", 10 ] ] ],
     "components": [
-      [ [ "milk_standard", 2, "LIST" ] ],
+      [ [ "milk_standard", 2, "LIST" ], [ "lf_milk_standard", 2, "LIST" ] ],
```
Comment (kevingranade) [inline data/json/recipes/food/frozen.json:179]: This is possible but requires another source of fat.
https://www.masterclass.com/articles/almond-milk-ice-cream-recipe#3-ways-to-customize-almond-milk-ice-cream

File: data/json/recipes/food/frozen.json [hunk 2]
```diff
@@ -238,7 +244,7 @@
     "book_learn": [ [ "mag_cooking", 0 ] ],
     "components": [
       [ [ "water_clean", 2 ] ],
-      [ [ "milk_standard", 2, "LIST" ] ],
+      [ [ "milk_standard", 2, "LIST" ], [ "lf_milk_standard", 2, "LIST" ] ],
```
Comment (kevingranade) [inline data/json/recipes/food/frozen.json:247]: As above, needs a fat source.

File: data/json/recipes/food/frozen.json [hunk 3]
```diff
@@ -335,7 +341,7 @@
     "qualities": [ { "id": "COOK", "level": 3 } ],
     "tools": [ [ [ "rock_quern", -1 ], [ "clay_quern", -1 ], [ "mortar_pestle", -1 ], [ "food_processor", 10 ] ] ],
     "components": [
-      [ [ "milk_standard", 2, "LIST" ] ],
+      [ [ "milk_standard", 2, "LIST" ], [ "lf_milk_standard", 2, "LIST" ] ],
```
Comment (kevingranade) [inline data/json/recipes/food/frozen.json:344]: This one is fine due to the eggs.

File: data/json/recipes/recipe_food.json [hunk 4]
```diff
@@ -3567,7 +3572,7 @@
     "tools": [ [ [ "surface_heat", 8, "LIST" ] ] ],
     "components": [
       [ [ "flour_any", 20, "LIST" ] ],
-      [ [ "milk_standard_raw", 1, "LIST" ] ],
+      [ [ "milk_standard_raw", 1, "LIST" ], [ "lf_milk_standard", 1, "LIST" ] ],
```
Comment (kevingranade) [inline data/json/recipes/recipe_food.json:3575]: Need a fat source.
https://www.simplywhisked.com/drop-biscuits/

File: data/json/recipes/recipe_food.json [hunk 5]
```diff
@@ -4094,7 +4099,12 @@
     "tools": [ [ [ "surface_heat", 5, "LIST" ] ] ],
     "components": [
       [ [ "hickory_nut_shelled", 4 ] ],
-      [ [ "milk_standard_raw", 1, "LIST" ], [ "milk_powder", 1 ], [ "con_milk", 1 ] ],
+      [
+        [ "milk_standard_raw", 1, "LIST" ],
+        [ "lf_milk_standard", 1, "LIST" ],
```
Comment (kevingranade) [inline data/json/recipes/recipe_food.json:4104]: Need a source.

Comment (AtomicFox556) [reply data/json/recipes/recipe_food.json:4104]: https://www.tyrantfarms.com/recipe-how-to-make-hickory-nut-ambrosia/

Recipe says to add it at the very end or just let people add milk according to their taste. Supposedly if someone wouldn't want to use actual milk, they could add lactose-free milk instead.

**Comment by AtomicFox556** [comment]
I've decided to just remove recipe changes from the recipes that would need a fat source, as it'd require a new recipe at this point.

**Comment by ccaviness** [comment]
I know it's not supposed to fix it, but just to be sure I checked and food created with these are still flagged as allergic for Lactose Intolerance.  See #48489 for an older issue about this.

**Pull request merged**